docs/design decisions/settings.md

```
@@ -0,0 +1,67 @@
+## Tenant Settings
+Right now all settings are bundled up together in a settings table
+
+## High level Idea
+- Each "Setting" will be a nested JSON value (at proto API level it is a a 'Struct').
+- Each setting has a name
+
+examples :
+- "provider.settings" (<- name) : (value ->)  { "openai":{}, "google":{}   }
+
+API for getting and setting these settings - 
+GetSetting(name) proto.Struct (https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.proto)
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)